pkg/testing/fixture.go

```diff
@@ -169,6 +170,25 @@ func (f *Fixture) Prepare(ctx context.Context, components ...UsableComponent) er
 	return nil
 }
 
+// Configure replaces the default Agent configuration file with the provided
+// configuration. This must be called after `Prepare` is called but before `Run`
+// or `Install` can be called.
+func (f *Fixture) Configure(ctx context.Context, yamlConfig []byte) error {
```

Comment:
Do we want as a testing framework to provide the `yamlConfig` as a structure instead? Would provide type checking and easier to use framework in my opinion.

I know that means that we have to keep adding to the structure as we change things, but being its just in the testing framework that should be easy to change at will.

Reply:
Being that these are integration/E2E tests I'd prefer to keep it as a string, more because it's closer to what a user would do.  It also tests the configuration parsing/loading parts as a result.  

If these were unit tests, I'd agree that going with a structure (or reusing the one we already havein the Agent code) would be better.

Reply:
True.